### DISCUSSION TITLE
Add existing photos into Import Library

### DISCUSSION THREAD

**Discussion by patrickli** [immich-app/immich#6528] [Q&A] [answered]
I've installed Immich on a new self-built NAS running TrueNAS. I already have backups of the photos from my phone in one of the datasets following the `STORAGE_LABEL/YEAR/DATE` folder structure.

So my question is, can I point the library directory to the dataset and just tell Immich to scan/build everything from scratch, without using the CLI import tool? As far as I understand the CLI import tool will move/copy everything to the uploads folder and let Immich do the organising. But since all of my files are already organised, can I just manually run the "library" job or a combination of other jobs to make them appear in my upload library?

I don't want to use the external library for this as these are photos supposed to be uploaded from my phone anyway.

Thank you very much.

**Comment by schuhbacca** [comment, marked as answer]
You cannot. You will either need to use the external library or bulk upload them through the CLI to get them into the default library.

**Reply by patrickli** [reply]
I see. I actually looked at the source code after posting this and the cli is just calling the upload API with raw image data over the network.
